### PR TITLE
do not use None as key in local_dict

### LINKED ISSUE #22822: `parse_expr` evaluates `is None` expression to False since 1.9
Since version 1.9 `parse_expr`  evaluates a `is None` expression to False instead of True (as I would expect) when logic_data passed contains a  a None value.

```python
from sympy.parsing.sympy_parser import parse_expr

logic = "some_parameter is None"
data = {'some_parameter': None}
print(parse_expr(logic, data))  # evaluates to True in 1.8 and False in 1.9
data = {'some_parameter': 'got it'}
print(parse_expr(logic, data))  # evaluates to False
```

Expected result is True, False and not False, False

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

fixes #22822
#### Release Notes

* parsing
    * corrected parsing error that happened when when local_dict value was None

### CODE REVIEW

**Comment by oscargus** [comment]
Closing and reopening as the required tests have changed.

File: sympy/parsing/sympy_parser.py [hunk 1]
```diff
@@ -20,6 +20,8 @@
 from sympy.functions.elementary.miscellaneous import Max, Min
 
 
+null = ''
```
Comment (ThePauliPrinciple) [inline sympy/parsing/sympy_parser.py:23]: It feels to me that using an empty string is going to make this vulnerable to (hard to debug) bugs in the future. (when for some reason `''` ends up as a key for another reason)

```python
class ParserNull:
    pass
null = ParserNull()
```
Might be a bit safer, but feel free to ignore if you don't think it is needed here.

Comment (smichr) [reply sympy/parsing/sympy_parser.py:23]: I understand the sentiment...but when would we ever parse a null string which, by definition, has nothing to parse? There is also a test that `null` does not appear in `local_dict` so if it ever does we could revisit it then and make the change in one place (with what you suggested). I'll make the change, however.

Comment (smichr) [inline sympy/parsing/sympy_parser.py:23]:
```suggestion
class ParserNull:
    pass
null = ParserNull()
```

File: sympy/parsing/sympy_parser.py [hunk 2]
```diff
@@ -1046,6 +1050,8 @@ def parse_expr(s, local_dict=None, transformations=standard_transformations,
         local_dict = {}
     elif not isinstance(local_dict, dict):
         raise TypeError('expecting local_dict to be a dict')
+    elif null in local_dict:
+        raise ValueError('cannot use "" in local_dict')
```
Comment (smichr) [inline sympy/parsing/sympy_parser.py:1054]:
```suggestion
        raise ValueError('cannot use ParserNull as local_dict key')
```

File: sympy/parsing/sympy_parser.py (hunk 1 above)
Comment (smichr) [reply sympy/parsing/sympy_parser.py:23]: On second thought, let's leave it as is -- by definition nobody should be changing nothing to something.

**Pull request merged**